### DISCUSSION TITLE
How to select all columns in a file

### DISCUSSION THREAD

**Discussion by marcianoemo** [grafana/grafana-csv-datasource#63] [Q&A] [answered]
Thank you for your contribution.
I have a file that is populated with random information. New columns are created at all times. How can I select all columns to create a line graph?

For exemple:
date,metric1,metric2,metric3,...,metric51,metric52,...,
2021-04-29 00:09:00,10,20,15,...,08,12,...,
2021-04-29 00:10:00,11,15,16,...,08,07,...,
2021-04-29 00:11:00,09,10,16,...,08,05,...,
2021-04-29 00:12:00,13,08,20,...,10,06,...,
2021-04-29 00:13:00,14,12,15,...,05,05,...,

My application creates metric## automatically and a new column is added to the .csv file.

**Comment by marcusolsson** [comment, marked as answer]
Hm. You can only graph it with a valid schema (a time field and number fields). Since the columns seem to be dynamic, I guess you'd have to automatically configure the schema. This is not supported unfortunately and I don't have a clear solution in mind. Maybe we could support regex in the schema editor? So you'd set `metric[0-9]+` as the field name, and then set the type.

**Reply by aboehm64** [reply]
the regex as the field name to configure the type of the field would be perfect for me. i have the same problem as described above.
